### ISSUE TITLE
WSL2 support?

### ISSUE THREAD

**Issue by oscarbg** [wight554/ClearWSL#2]

**Comment by wight554** [comment]
can't test until https://github.com/microsoft/WSL/issues/4371 fixed

**Comment by oscarbg** [comment]
I might be lucky on WSL side as both a Ubuntu and Debian installed from "Win Store" as WSL1 and WSL2 respectively work on my 18956 insider build..
Installed Clear as WSL1 (setting default wsl version to 1) and it worked without problems also..
but converting the machine to WSL2 via via: "wsl --set-version Clear 2" failed miserably (seems towards the end of conversion as some 1GB vhdx was being created before failing and deleting that file)
I got warnings about some files which were correct links to other files ( I mean like ln -sf a b) from Clear not being found in the conversion process.. like some man files /usr/share/man3/.../CN_ZK/..

**Comment by wight554** [comment]
I linked that issue not just for reference
It's clearly written here that all distros with /usr/bin => /bin symlink fail to start wsl2
And clear linux is one of those

**Comment by oscarbg** [comment]
ok sorry.. didn't read thoroughly..

**Comment by Biswa96** [comment]
There is no special criteria for WSL2. I tested this repository package in WSL2 build 18945 and it works. There is two choice -- 1. Wait for next 2-4 weeks for next insider builds or 2. Revert back to build 18945 to test WSL2.

**Comment by wight554** [comment]
fixed by https://github.com/wight554/ClearWSL/commit/313136860cb6923747871ef60f476cb8b58ab44f

**Comment by wight554** [comment]
nvm that hack didn't work fine here, at least the way i did it
workaround described here https://github.com/microsoft/WSL/issues/4371#issuecomment-521107556 won't work too
tldr swupd doesn't like this

**Comment by shayne** [comment]
Updated to 18970. Ran `.\Clear.exe` with default WSL set to version as 1.

Converted to 2 via `wsl.exe --set-version <distro-name> 2`.

Works.

**Comment by wight554** [comment]
Tnx for confirmation, haven't updated yet, closing the issue
